share/p2p/peers/manager.go

```diff
@@ -418,6 +428,31 @@ func (m *Manager) isBlacklistedPeer(peerID peer.ID) bool {
 	return !m.connGater.InterceptPeerDial(peerID)
 }
 
+func (m *Manager) arePeerProtocolsCompatible(peerID peer.ID) bool {
+	peerProtos, _ := m.host.Peerstore().GetProtocols(peerID)
```

Comment:
There is one more method in the peerStore that fits better in this case - `SupportsProtocols`

Reply:
Thanks!